File: Python Interview Preparation/Educative/pythonProject/BinarySearch/BinarySearch.py
def binary_search(nums, target):
    # Replace this placeholder return statement with your code
    low = 0
    high = len(nums) - 1

    while low <= high:
        mid = low + (high - low) // 2
        if nums[mid] == target:
            return mid
        elif nums[mid] <= target:
            low = mid + 1
        else:
            high = mid - 1 

    return -1


print(binary_search([-30, -27, -8, -6, -1] , -1))

File: LeetCode Problems/src/BinarySearch.java

public class BinarySearch {

	public static void main(String[] args) {

		int[] nums = { -1, 0, 3, 5, 9, 12 };
		int target = 9;
		int start = 0, end = nums.length - 1, mid = 0;
		
		while (start <= end) {
			mid = start + (end - start) / 2;
			if (nums[mid] == target) {
				System.out.print("Found");
				break;
			} else if (nums[mid] > target) {
				end = mid - 1;
			} else {
				start = mid + 1;
			}
		}
		System.out.print("Mis id "+mid);

	}

}
